internal/pkg/status/legacy_status.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	clientu "sigs.k8s.io/cli-experimental/internal/pkg/client/unstructured"
+)
+
+// IsReadyFn - status getter
+type IsReadyFn func(*unstructured.Unstructured) (bool, error)
```

Comment:
Rather than return "isReady", should these return synthetic conditions which can be used by the generalized libraries?

Reply:
Thats a good idea. Let me try some out in a branch.

Reply:
Yes that sounds wonderful. And helps in handling complete, ready, settled cases.

Reply:
added conditions.